### PR TITLE
Address review comments from #98259

### PR DESCRIPTION
**Pull request by jyn514**
It got approved so fast I didn't have time to make changes xD

r? @eholk

### CODE REVIEW

**Comment by jyn514** [comment]
@eholk are you not part of the org? you should join compiler-contributors :)

anyway, I've assigned esteban, but happy to approve this on your behalf if you have time

@bors delegate=eholk

**Comment by eholk** [comment]
It looks like this includes the patch from #98259, right? I only reviewed the patch on top of that one that addressed review comments since the other PR was already approved.

@bors r+

**Comment by eholk** [comment]
> @eholk are you not part of the org? you should join compiler-contributors :)

I'd love to! What's the process for doing this?

**Comment by jyn514** [comment]
Looks like these are the latest guidelines: https://forge.rust-lang.org/compiler/membership.html#how-promotion-decisions-are-made
I'll send an email this afternoon :)

**Comment by eholk** [comment]
Awesome, thanks!

File: compiler/rustc_middle/src/ty/context.rs [hunk 1]
```diff
@@ -2224,6 +2224,20 @@ impl<'tcx> TyCtxt<'tcx> {
         })
     }
 
+    /// Given a `ty`, return whether it's an `impl Future<...>`.
+    pub fn ty_is_opaque_future(self, ty: Ty<'_>) -> bool {
```
Comment (estebank) [inline compiler/rustc_middle/src/ty/context.rs:2228]: I'm pretty sure this can be used in multiple places! I recall seeing similar needs elsewhere.

**Review by estebank — APPROVED** [review]

File: compiler/rustc_middle/src/ty/context.rs (hunk 1 above)
Comment (jyn514) [reply compiler/rustc_middle/src/ty/context.rs:2228]: Hmm, this is the only other place I see looking at opaque Future types:
https://github.com/rust-lang/rust/blob/75337775f776250a3a29c951344186e698c11c75/compiler/rustc_infer/src/infer/error_reporting/mod.rs#L1834
that needs the `Output` type from the future - maybe I shouldn't have added this method and I should just use `get_impl_future_output_ty().is_some()` instead?

Comment (estebank) [reply compiler/rustc_middle/src/ty/context.rs:2228]: Yes, that'd be a good idea. Could you make a small PR for it at some point?

**Pull request merged**